cmd/arduino-app-cli/version/version.go

```diff
@@ -16,34 +16,132 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
 
 	"github.com/spf13/cobra"
 
 	"github.com/arduino/arduino-app-cli/cmd/feedback"
+	"github.com/arduino/arduino-app-cli/cmd/i18n"
 )
 
-func NewVersionCmd(version string) *cobra.Command {
+// The actual listening address for the daemon
+// is defined in the installation package
+const (
+	DefaultHostname = "localhost"
+	DefaultPort     = "8800"
+)
+
+func NewVersionCmd(clientVersion string) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "version",
-		Short: "Print the version number of Arduino App CLI",
+		Short: "Print the client and server version numbers for the Arduino App CLI.",
 		Run: func(cmd *cobra.Command, args []string) {
-			feedback.PrintResult(versionResult{
-				AppName: "Arduino App CLI",
-				Version: version,
-			})
+			host, _ := cmd.Flags().GetString("host")
+
+			versionHandler(clientVersion, host)
 		},
 	}
+	cmd.Flags().String("host", fmt.Sprintf("%s:%s", DefaultHostname, DefaultPort),
+		"The daemon network address [host]:[port]")
 	return cmd
 }
 
-type versionResult struct {
-	AppName string `json:"appName"`
+func versionHandler(clientVersion string, host string) {
+	httpClient := http.Client{
+		Timeout: time.Second,
+	}
+	result := doVersionHandler(httpClient, clientVersion, host)
+	feedback.PrintResult(result)
+}
+
+func doVersionHandler(httpClient http.Client, clientVersion string, host string) versionResult {
+	url, err := getValidOrDefaultUrl(host)
+	if err != nil {
+		feedback.Fatal(i18n.Tr("Error: invalid host:port format"), feedback.ErrBadArgument)
+	}
```

Comment:
Here you can just use `url.Parse`, but anyway I would parse the url right after getting the flag, e.g., `host, _ := cmd.Flags().GetString("host")`, so you can directly handle the error. 
Then here you can just add the path with 
```
url = url.Join("/v1/version")